### ISSUE TITLE
Retire the tray/system icon on Linux

### ISSUE THREAD

**Issue by forteller** [ramboxapp/community-edition#1118]
Gnome is [retiring the legacy system tray](http://www.omgubuntu.co.uk/2017/09/will-you-miss-gnome-legacy-tray). Ubuntu removed the same thing a long time ago and is now moving to Gnome. When I open Rambox it opens only as a system icon in the tray, and I have to right click on it and click "Show/hide Rambox" – This will soon no longer be possible.

I hope you will be able to switch the new system before Ubuntu 17.10 arrives. Thank you so much!

**Comment by SevereOverfl0w** [comment]
Reading the article linked, I cannot determine what the "new system" is, it seems that there is no tray?

There are plenty of other Linux setups that will retain the tray, so I think any "retirement" should involve avoiding the tray in cases where there isn't anything listening to the tray event.

**Comment by chrpinedo** [comment]
This is the link to the migration guidelines that GNOME provides: https://wiki.gnome.org/Initiatives/StatusIconMigration/Guidelines

As far as I understand, the idea would be to launch always the primary window of the application when the application is launched, if the application is already running in background it should show the current instance of rambox (no a new one). The application should make use of GNOME notification system (it is possible to set one specific action per notification so if the user clicks one whatsapp notification, the application is opened in the specific whatsapp conversation). In case of closing the primary window the application should run in background if it is required and notifications should be the only way that an application interacts with the user while it runs in the background.

**Comment by sylvainar** [comment]
Yup, I just switched to Ubuntu 17.10 and here I am, facing this problem. It will become more and more critical, because people will start upgrading.

**Comment by sylvainar** [comment]
Update : So the Ubuntu / Gnome default configuration seems to have its tray, and it seems to work correctly because of the Steam icon, but no Rambox. I've no idea why.
![screenshot](https://user-images.githubusercontent.com/9823286/31863506-84423a24-b74e-11e7-966a-465e131d3727.png)

**Comment by philb61** [comment]
Based on the article inked above, I installed the TopIconsPlus Gnome Shell extension (just typed that in the search box from "all applications"), and installing that brings the Rambox and other icons back.

**Comment by forteller** [comment]
There seems to be two ways of showing system tray icons. Some, like Steam, are shown if you have the Kstatusnotifieritem/appindicator support extension (or Canonicals fork Ubuntu appindicators) installed, while others are shown if you have the Topicons plus extension installed. Unfortunately only the first one is installed by default in Ubuntu 17.10. Maybe it would be a good idea to switch to that type of tray icon to avoid confusion and irritation by those not in the know about this? I don't know how much work that would be, though.

AFAIK they want to retire both kinds of tray icons, they've just postponed it a bit to give devs more time. So maybe it's just a waste of time to change the type of tray icon and you should rather spend that time figuring out how to do the new system they want devs to use? I'm no dev, so I don't know anything… :)

**Comment by sylvainar** [comment]
Seems right. But what is the expected behavior on Gnome ? For the notifications, you can use the embedded Gnome notifications system, but you just let the app run in background with no way to know that it's still working ?

**Comment by adambro** [comment]
It seems that appindicator is not fully supported yet in elector, see https://github.com/electron/electron/issues/10427

**Comment by chrpinedo** [comment]
> Seems right. But what is the expected behavior on Gnome ? For the notifications, you can use the embedded Gnome notifications system, but you just let the app run in background with no way to know that it's still working ?

According to https://wiki.gnome.org/Initiatives/StatusIconMigration/Guidelines, that is the idea. IMHO it is something similar to Android phone where you only has notifications but not indications of which applications are running.

**Comment by MrObvious** [comment]
It sounds like it becomes more a service. Maybe just make an option to where it runs in the background but disappears unless you execute rambox again or open a notification?

**Comment by MrObvious** [comment]
How hard would this be to implement? I'm not getting notifications on Gnome on 17.10.

**Comment by fvulich** [comment]
Does this issue still exists in v0.7.5?

**Comment by chrpinedo** [comment]
Yes, GNOME removed system tray icons and it suggests a new method to interact with background running applications as it detailed in https://wiki.gnome.org/Initiatives/StatusIconMigration/Guidelines. 

My solution is to run rambox with: "Display behaviour:" "Show in Taskbar" and "When closing the main window:" "Keep in tray". So I have the rambox application open in one desktop every time.

Another solution would be to install or recommend to install the "Kstatusnotifieritem/appindicator support" extension from gnome extensions. This GNOME extension recovers system trays in the GNOME desktop.

However, IMHO this is a design decision to take. It is worth converting rambox compatible with vanilla GNOME or you recommend any of the previous solutions.

**Comment by chrpinedo** [comment]
so many comments and links that seem to be worthless...

**Comment by forteller** [comment]
I don't understand why this would be closed just because I haven't responded. Other people have responded fine, so why would there be some sort of demand that I respond? I'm no coder or Gnome expert, I just wanted to give a heads up about the new system that would be coming… in 2017…
